pkg/scheduler/plugins/deviceshare/allocator_gpu.go

```diff
@@ -114,6 +122,63 @@ func removeZeroDevice(originalResources deviceResources) deviceResources {
 	return refinedResources
 }
 
+func (a *GPUAllocator) generalAllocate(requestCtx *requestContext, nodeDevice *nodeDevice, desiredCount int, maxDesiredCount int, allocateContext *AllocateContext) ([]*apiext.DeviceAllocation, *framework.Status) {
+	allocations, status := allocateByDeviceTopology(requestCtx.gpuRequirements, nodeDevice.gpuTopologyScope, allocateContext)
+	if !status.IsSuccess() {
+		return nil, status
+	}
+	// if same NUMANode or PCIE are not required, allocateByDeviceTopology may return (nil, nil), so we should check if allocations is nil
+	if len(allocations) != 0 {
+		return allocations, nil
+	}
+
+	return defaultAllocateDevices(nodeDevice, requestCtx, requestCtx.gpuRequirements.requestsPerGPU, desiredCount, maxDesiredCount, schedulingv1alpha1.GPU, nil)
+}
+
+func (a *GPUAllocator) allocateByTemplate(requestCtx *requestContext, nodeDevice *nodeDevice, desiredCount int, maxDesiredCount int, allocateContext *AllocateContext) ([]*apiext.DeviceAllocation, *framework.Status) {
+	if !requestCtx.gpuRequirements.gpuShared {
+		return nil, nil
+	}
+	vendor, model := requestCtx.node.Labels[apiext.LabelGPUVendor], requestCtx.node.Labels[apiext.LabelGPUModel]
+	if vendor != apiext.GPUVendorHuawei {
+		return nil, nil
+	}
+
+	requestsPerGPU := requestCtx.gpuRequirements.requestsPerGPU
+	if len(requestsPerGPU) > 1 {
+		// koord style: match a template for accurate resources specified by user
+		templates, err := a.gpuSharedResourceTemplatesCache.getTemplates(vendor, model)
+		if err != nil {
+			return nil, framework.NewStatus(framework.Error, err.Error())
+		}
+		var templateName string
+		for name, template := range templates {
+			if quotav1.Equals(requestsPerGPU, template) {
+				templateName = name
+				break
+			}
+		}
+		if templateName == "" {
```

Comment:
Can this logic be placed in the Prefilter stage to reduce the performance overhead of node-level allocation?

Reply:
No, because different nodes with different GPU models may have different templates, so the template match can only be done at node level.

Reply:
I know, but some pre-computation is possible. For example, can we calculate the template name that can be matched for each vendor-model in advance according to the Pod application in PreFilter? Here, we only need to map the template name according to the vendor-model?